estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,88 @@
+import math
+
+from dateutil import relativedelta
+
+from odoo import _, api, exceptions, fields, models
+
+
+class EstatePropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Estate property Offer'
+    _order = 'price desc'
+
+    price = fields.Float(string='Price')
+    status = fields.Selection(selection=[('accepted', 'Accepted'), ('refused', 'Refused')], copy=False)
+    partner_id = fields.Many2one('res.partner', string='Buyer', required=True)
+    property_id = fields.Many2one('estate.property', string='Property', required=True)
+    validity = fields.Integer(string='Validity (days)', default=7)
+    date_deadline = fields.Date(string='Deadline', compute='_compute_deadline', inverse='_inverse_deadline')
+    property_type_id = fields.Many2one(
+        string='Property Type',
+        related='property_id.property_type_id',
+        store=True,
+    )
+
+    _sql_constraints = [
+        ('check_price', 'CHECK(price > 0)', 'The price must be strictly positive.'),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        properties = {
+            p.id: p
+            for p in self.env['estate.property'].browse([
+                e.get('property_id') for e in vals_list if e.get('property_id')
+            ])
+        }
+
+        for vals in vals_list:
+            property = properties.get(vals.get('property_id'), '')
+
+            if not properties or 'price' not in vals:
+                continue
+
+            if vals['price'] < min([*property.offer_ids.mapped('price'), math.inf]) and len(property.offer_ids) > 1:
```

Comment:
Several points here:
- The check on the length can be done first;
- The new offer is not there yet, so the check should be: is there already another offer? (0, not 1)
- There is a best_price field that does the check on offers already, meaning the check on offers is useless;
- Following my previous comment, I'd check the price here only
```suggestion
            if vals.get('price') and vals['price'] < property.best_price:
```
You could also use the Walrus operator if you're into it
```suggestion
            if (price := vals.get('price')) and price < property.best_price:
```